thrust/thrust/detail/type_traits.h

```diff
@@ -60,12 +60,15 @@ namespace detail
      constexpr __host__ __device__ value_type operator()() const noexcept { return value; }
    };
 
- /// typedef for true_type
+ /// typedef for true_type, delete when no longer needed?
  typedef integral_constant<bool, true>  true_type;
 
- /// typedef for true_type
+ /// typedef for true_type, delete when no longer needed?
  typedef integral_constant<bool, false> false_type;
 
+ using true_types = cuda::std::true_type; // avoid friction with rest of codebase
+ using false_types = cuda::std::false_type; // better name
+
```

Comment:
I believe the issue is that `thrust::integral_constant` is different from `cuda::std::integral_constant`

I would suggest to simply defer that change until you replace that type with some aliases